### PR TITLE
GitHub actions release

### PR DESCRIPTION
**Pull request by mattwylder**
### Summary of changes
 - Added Release workflow to GitHub actions

### Checklist
~- [ ] Added a changelog entry~

- @sshropshire

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -40,3 +44,41 @@ nexusStaging {
...
+        def gradleFile = new File('build.gradle')
+        def gradleFileText = gradleFile.text.replaceFirst("\\nversion '\\d+\\.\\d+\\.\\d+(-.*)?'", "\nversion '" + versionParam + "'")
+        gradleFile.write(gradleFileText)
+    }
+}
+
+task changeREADMEVersion {
+    doLast {
+        def gradleFile = new File('README.md')
+        def gradleFileText = gradleFile.text.replaceFirst("com.braintreepayments.api:drop-in:\\d+\\.\\d+\\.\\d+(-.*)?'", "com.braintreepayments.api:drop-in:" + versionParam + "'")
```
Comment (mattwylder) [inline build.gradle:59]: Just noticed I need to fix this

File: build.gradle [hunk 2]
```diff
@@ -40,3 +44,41 @@ nexusStaging {
...
+        gradleFile.write(gradleFileText)
+    }
+}
+
+task incrementSNAPSHOTVersion {
```
Comment (mattwylder) [inline build.gradle:64]: Do we do snapshot versioning the same way as in other repos?

Comment (sshropshire) [reply build.gradle:64]: This is legacy I think for the most part we stopped providing SNAPSHOT builds. On the other hand, we should consider bringing them back to allow merchants to try out unreleased features. It's probably good to have in the build script for now.

**Review by sshropshire — APPROVED** [review]
🌊

File: .github/actions/set_github_user/action.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: Set GitHub User
+description: 'Sets github user for a particular workflow'
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        git config user.name paypalsdk
+        git config user.email sdks@paypal.com
```
Comment (jaxdesmarais) [inline .github/actions/set_github_user/action.yml:8]: I think we decided we should use braintreeps for btsdks and paypalsdk for unified sdk

```suggestion
        git config user.name braintreeps
        git config user.email code@getbraintree.com
```

**Review by jaxdesmarais — APPROVED** [review]
Just 1 comment around updating the git config to use the correct name an email but otherwise looks good

**Pull request merged**